internal/services/netapp/netapp_backup_policy_resource.go

```diff
@@ -0,0 +1,331 @@
+			}
+
+			err = client.BackupPoliciesCreateThenPoll(ctx, id, parameters)
+			if err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+
+			return nil
+		},
+	}
+}
+
+func (r NetAppBackupPolicyResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 120 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.NetApp.BackupPolicyClient
+
+			id, err := backuppolicy.ParseBackupPolicyID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			metadata.Logger.Infof("Decoding state for %s", id)
+			var state netAppModels.NetAppBackupPolicyModel
+			if err := metadata.Decode(&state); err != nil {
+				return err
+			}
+
+			var shouldUpdate = false
+
+			update := backuppolicy.BackupPolicyPatch{
+				Properties: &backuppolicy.BackupPolicyProperties{},
+			}
+
+			// Checking properties with changes
+			if metadata.ResourceData.HasChange("tags") {
+				update.Tags = pointer.To(state.Tags)
+				shouldUpdate = true
+			}
+
+			if metadata.ResourceData.HasChange("daily_backups_to_keep") {
+				update.Properties.DailyBackupsToKeep = pointer.To(state.DailyBackupsToKeep)
+				shouldUpdate = true
+			}
+
+			if metadata.ResourceData.HasChange("weekly_backups_to_keep") {
+				update.Properties.WeeklyBackupsToKeep = pointer.To(state.WeeklyBackupsToKeep)
+				shouldUpdate = true
+			}
+
+			if metadata.ResourceData.HasChange("monthly_backups_to_keep") {
+				update.Properties.MonthlyBackupsToKeep = pointer.To(state.MonthlyBackupsToKeep)
+				shouldUpdate = true
+			}
+
+			if metadata.ResourceData.HasChange("enabled") {
+				update.Properties.Enabled = pointer.To(state.Enabled)
+				shouldUpdate = true
+			}
+
+			if shouldUpdate {
+				metadata.Logger.Infof("Updating %s", id)
+
+				if err := client.BackupPoliciesUpdateThenPoll(ctx, pointer.From(id), update); err != nil {
+					return fmt.Errorf("updating %s: %+v", id, err)
+				}
+
+				metadata.SetID(id)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r NetAppBackupPolicyResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+
+			client := metadata.Client.NetApp.BackupPolicyClient
+
+			id, err := backuppolicy.ParseBackupPolicyID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			metadata.Logger.Infof("Decoding state for %s", id)
+			var state netAppModels.NetAppBackupPolicyModel
+			if err := metadata.Decode(&state); err != nil {
+				return err
+			}
+
+			existing, err := client.BackupPoliciesGet(ctx, pointer.From(id))
+			if err != nil {
+				if existing.HttpResponse.StatusCode == http.StatusNotFound {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retrieving %s: %v", id, err)
+			}
+
+			if model := existing.Model; model != nil {
+				state.Location = location.NormalizeNilable(pointer.To(model.Location))
```

Comment:
We can simplify this
```suggestion
				state.Location = location.Normalize(model.Location)
```